server/src/main/java/org/apache/druid/server/coordinator/rules/PeriodBroadcastDistributionRule.java

```diff
@@ -65,6 +65,13 @@ public boolean appliesTo(Interval interval, DateTime referenceTimestamp)
     return Rules.eligibleForLoad(period, interval, referenceTimestamp, includeFuture);
   }
 
+  @Override
+  public Interval getEligibleInterval(DateTime referenceTimestamp)
+  {
+    return includeFuture ? new Interval(referenceTimestamp.minus(period), referenceTimestamp.plus(period))
```

Comment:
When `includeFuture` is true, we include everything in the future, not just upto the `period`.